src/Workspaces/CSharp/Portable/LanguageServices/CSharpSyntaxFactsService.cs

```diff
@@ -1817,7 +1808,22 @@ public bool IsOnTypeHeader(SyntaxNode root, int position)
 
         public bool IsInPropertyDeclarationHeader(SyntaxNode node)
         {
-            var propertyDeclaration = node.GetAncestor<PropertyDeclarationSyntax>();
+            var propertyDeclaration = node.GetAncestorOrThis<PropertyDeclarationSyntax>();
+            if (propertyDeclaration == null)
+            {
+                return false;
+            }
+
+            var start = propertyDeclaration.AttributeLists.LastOrDefault()?.GetLastToken().GetNextToken().SpanStart ??
+                        propertyDeclaration.SpanStart;
+            var end = propertyDeclaration.Identifier.FullSpan.End;
+
+            return node.Span.Start >= start && node.Span.End <= end;
```

Comment:
i feel like most of this logic can be extracted into a helper that says `IsInHeader(node, propertyDecl, propertyDecl.PrefferedEndChild)`.   It would find the part after the attributes for `node` as the `start` and would use the `end` passed in.  We can then share this uniformly.

Reply:
Where exactly would PrefferedEndChild come from? Extension method (if so, where should it live?) / having it as a Func<SyntaxNode, SyntaxTokenOrNode> extractor func? (if so, where should they live?)

Reply:
it would be determined on a case by case bases.  so IsOnPropertyHeader would hardcode in whatever it thought was appropriate (maybe `.Identifier` for C# and the `AsClause` for VB?).

Reply:
Nvm, I see your point now.

Reply:
There's still the issue with `propertyDecl.PrefferedEndChild` not always being one, sometimes it's one if it's null or some other... 

Know how to solve it but won't be as pretty.